src/AvenueClothing/Controllers/BasketpreviewController.cs

```diff
@@ -102,11 +101,17 @@ private PurchaseOrderViewModel MapOrder()
         [HttpPost]
         public ActionResult Index()
         {
-            TransactionLibrary.RequestPayments();
+            var payment = TransactionLibrary.GetBasket().Payments.First();
```

Comment:
Would we not want to get the last payment? If there's more than one, I assume the prior ones failed?

Reply:
Create payment defaults to override existing created payments, thus there should always only be one - unless there's a more complex flow with more payments or you are using the gift card app, in which you need to narrow the payments even further. I think it is out of scope of avenue clothing since it only supports a single payment option.